crypto/hpke/hpke_highlevel.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* An OpenSSL-based high level interface to HPKE */
+#include <stddef.h>
+#include <openssl/hpke.h>
+#include <openssl/core_names.h>
+#include <openssl/crypto.h>
+#include <internal/packet.h>
+#include <openssl/err.h>
+
+int OSSL_HPKE_BOX_seal(OSSL_LIB_CTX *libctx, const char *propq,
```

Comment:
Maybe put libctx, propq last in the params..
They are kind of optional parameters just used for algorithm fetching.

Reply:
That would align them with OSSL_HPKE_CTX_new, which makes sense. However a bunch of other functions seem to take libctx first and propq last, like EVP_CIPHER_fetch, EVP_PKEY_CTX_new_from_name, so I'm not sure if we'd be making an inconsistency worse by doing that. I'm leaning toward aligning with OSSL_HPKE_CTX_new.

Reply:
The reason they are like that in the fetch is that libctx is an important parameter...

Reply:
The grease API got it wrong though...